src/banks/registries/directory.py

```diff
@@ -38,24 +44,69 @@ def _load(self):
     def _scan(self):
         self._index: PromptFileIndex = PromptFileIndex()
         for path in self._path.glob("*.jinja*"):
-            pf = PromptFile(name=path.stem, version="0", path=path)
+            name, version = path.stem.rsplit(".", 1) if "." in path.stem else (path.stem, DEFAULT_VERSION)
+            pf = PromptFile(name=name, version=version, path=path, meta={})
             self._index.files.append(pf)
         self._index_path.write_text(self._index.model_dump_json())
 
-    def get(self, *, name: str, version: str | None = None) -> "Prompt":
-        version = version or "0"
+    def get_prompt(self, *, name: str, version: str | None = None) -> "Prompt":
+        version = version or DEFAULT_VERSION
         for pf in self._index.files:
             if pf.name == name and pf.version == version and pf.path.exists():
                 return Prompt(pf.path.read_text())
         raise TemplateNotFoundError
 
-    def set(self, *, name: str, prompt: Prompt, version: str | None = None, overwrite: bool = False):
-        version = version or "0"
+    def _get_prompt_file(self, *, name: str, version: str) -> PromptFile | None:
         for pf in self._index.files:
-            if pf.name == name and pf.version == version and overwrite:
-                pf.path.write_text(prompt.raw)
-                return
-        new_prompt_file = self._path / "{name}.{version}.jinja"
+            if pf.name == name and pf.version == version:
+                return pf
+        return None
+
+    def _create_pf(  # pylint: disable=too-many-arguments
+        self, *, name: str, prompt: Prompt, version: str, overwrite: bool, meta: dict
+    ) -> "PromptFile":
+        pf = self._get_prompt_file(name=name, version=version)
+        if pf:
+            if not overwrite:
+                msg = f"Prompt {name}.{version}.jinja already exists. Use overwrite=True to overwrite."
+                raise ValueError(msg)
+            pf.path.write_text(prompt.raw)
+            pf.meta = meta
+            return pf
+        new_prompt_file = self._path / f"{name}.{version}.jinja"
         new_prompt_file.write_text(prompt.raw)
-        pf = PromptFile(name=name, version=version, path=new_prompt_file)
-        self._index.files.append(pf)
+        pf = PromptFile(name=name, version=version, path=new_prompt_file, meta=meta)
+        return pf
+
+    def set(  # pylint: disable=too-many-arguments
+        self,
+        *,
+        name: str,
+        prompt: Prompt,
+        meta: dict | None = None,
+        version: str | None = None,
```

Comment:
Since strings are immutable, why not defaulting here directly? `version: str = DEFAULT_VERSION,`

Reply:
Makes sense, will do it.